teps/0131-tekton-conformance-policy.md

```diff
@@ -0,0 +1,268 @@
+
+
+## Glossary
+
+**Tekton Primitives**: are a set of basic data types in Tekton, providing common abstractions for describing and executing container-based, run-to-completion workflows, typically in service of CI/CD scenarios. There are three **Primitive Types**:
+* **Pipelines Primitives**: defines workloads execution pipeline
+  * Task, TaskRun, Pipeline, PipelineRun, CustomRun.
+* **Triggers Primitives**: defines events listening and handling
+  * Triggers, TriggerBindings.
+* **Results Primitives**: defines execution history of workloads defined by Pipeline Primitives.
+  * Results, Records
+
+**Tekton API Specification**: describes ***Tekton Primitives*** in detail.
+**Tekton GA API Specifications**: describes ***GA (V1 for now) Tekton Primitives*** in detail.
+
+**Tekton Pipelines API Specification**: describes ***Tekton Pipelines Primitives*** (`Task`, `TaskRun`, `Pipeline`, `PipelineRun`, `CustomRun` of all version tracks) in detail.
+**Tekton GA Pipelines API Specification**: describes ***Tekton GA Pipelines Primitives*** (`v1.Task`, `v1.TaskRun`, `v1.Pipeline`, `v1.PipelineRun`) in detail.
+
+**Tekton Conformance API Specification**: ***core Tekton GA API Specification*** each vendor MUST implement, fields existing in the Tekton API Specification but not in the Conformance API Specification are `OPTIONAL` for the Conformance Policy by default.
+**Tekton Pipelines Conformance API Specification**: ***core Tekton GA Pipelines API Specification*** each vendor MUST implement, including core fields of `v1.TaskRun` and `v1.PipelineRun`, within where `v1.Task` and `v1.Pipeline` can be incorporated by reference.
+
+**Tekton Users**: use ***Tekton Primitives*** to define CI/CD pipelines.
+**Tekton Vendors**: provide a product built upon Tekton that conforms to the ***Tekton Conformance Policy***.
+
+## Proposal
+
+Vendors can claim **Tekton `<Primitive Type>` Conformance** on the Services they provide. For example, a vendor can claim that its product conforms to "pipelines.v1.0" Conformance Policy, which is the initial version of Tekton Pipelines V1 Conformance Policy. See this section to know details about versioning.
+
+### Conformant Services
+
+Conformant Services are services that can serve **Tekton Conformance API Specifications**. We only define Conformance API Specifications for **Tekton GA Primitives**. For now, only **Tekton Pipelines Primitives** are eligible. *Worth noting that one prerequisite for conformance is that a primitive reaches GA, not every primitive that is GA is automatically part of conformance.*
+
+Vendors must be able to accept **Tekton `<Primitive Type>` Conformance API Specification**, defined in **YAML or JSON**, submitted **in any way** as long as the schema conforms to the policy[^schema_submission], then schedule workloads, produce outputs (resource conditions and results) as declared.
```

Comment:
Yeah that's true, if in the future, there are use cases, in order to enable the interoperability between different components, we may consider defining another level of the conformance policy -- HTTP API Conformance policy. I'd prefer to starting small and making additive changes on the Policy.

> Logs are problematic today. The way we handle logs in Tekton today is very k8s specific, and there is nothing in the API that points to how to retrieve logs - which is also problematic for custom tasks today.
Until that is solved, we won't be able to guarantee that a Tekton conformant service provides a way to retrieve logs

I assume by mentioning `logs` here, we are talking about `Logs` API instead of the log content right? 

For the `Log API`, what are the benefits if we include it in our conformance policy except for allowing `tkn` and `dashboard` to retrieve logs back?

Vendors may have their own way to do log management, if we limit the Log API, I'm afraid this may make vendor be less willing to conform to the policy.

Nevetheless, until we have a formed Log API, we may want to exclude it from the initial conformance policy version.

Reply:
Yes, until we have a log API, we cannot include it :) 
The lack of it will greatly harm interoperability and re-usability of tools like dashboard and tkn (as it already does for the integration of custom tasks) - so that's something we might want to consider in future. Not for this TEP though.

Reply:
Do we have an issue to track this? If not, I can open an issue for it.

Reply:
Sounds Good, thank you Andrea!